compiler/rustc_typeck/src/check/fn_ctxt/mod.rs

```diff
@@ -196,9 +202,23 @@ impl<'a, 'tcx> AstConv<'tcx> for FnCtxt<'a, 'tcx> {
                 self.param_env.caller_bounds().iter().filter_map(|predicate| {
                     match predicate.skip_binders() {
                         ty::PredicateAtom::Trait(data, _) if data.self_ty().is_param(index) => {
-                            // HACK(eddyb) should get the original `Span`.
-                            let span = tcx.def_span(def_id);
-                            Some((predicate, span))
+                            let trait_did = data.def_id();
```

Comment:
I don't think this change is needed. This code is used when type-checking the **body** of a function, and there can't (I think) be a cycle from something like supertraits or where-clauses to the body.